language/move-analyzer/src/bin/move-analyzer.rs

```diff
@@ -0,0 +1,164 @@
+        text_document_sync: Some(TextDocumentSyncCapability::Options(
+            TextDocumentSyncOptions {
+                open_close: Some(true),
+                // TODO: We request that the language server client send us the entire text of any
+                // files that are modified. We ought to use the "incremental" sync kind, which would
+                // have clients only send us what has changed and where, thereby requiring far less
+                // data be sent "over the wire." However, to do so, our language server would need
+                // to be capable of applying deltas to its view of the client's open files. See the
+                // 'move_analyzer::vfs' module for details.
+                change: Some(TextDocumentSyncKind::Full),
+                will_save: None,
+                will_save_wait_until: None,
+                save: Some(
+                    SaveOptions {
+                        include_text: Some(true),
+                    }
+                    .into(),
+                ),
+            },
+        )),
+        selection_range_provider: None,
+        hover_provider: None,
+        // The server provides completions as a user is typing.
+        completion_provider: Some(CompletionOptions {
+            resolve_provider: None,
+            // In Move, `foo::` and `foo.` should trigger completion suggestions for after
+            // the `:` or `.`
+            // (Trigger characters are just that: characters, such as `:`, and not sequences of
+            // characters, such as `::`. So when the language server encounters a completion
+            // request, it checks whether completions are being requested for `foo:`, and returns no
+            // completions in that case.)
+            trigger_characters: Some(vec![":".to_string(), ".".to_string()]),
+            all_commit_characters: None,
+            work_done_progress_options: WorkDoneProgressOptions {
+                work_done_progress: None,
+            },
+        }),
+        signature_help_provider: None,
+        definition_provider: None,
+        type_definition_provider: None,
+        implementation_provider: None,
+        references_provider: None,
+        document_highlight_provider: None,
+        document_symbol_provider: None,
+        workspace_symbol_provider: None,
+        code_action_provider: None,
+        code_lens_provider: None,
+        document_formatting_provider: None,
+        document_range_formatting_provider: None,
+        document_on_type_formatting_provider: None,
+        rename_provider: None,
+        document_link_provider: None,
+        color_provider: None,
+        folding_range_provider: None,
+        declaration_provider: None,
+        execute_command_provider: None,
+        workspace: None,
+        call_hierarchy_provider: None,
+        semantic_tokens_provider: None,
+        moniker_provider: None,
+        linked_editing_range_provider: None,
+        experimental: None,
```

Comment:
Nit, what about not setting any of the `None` fields and instead using the struct update syntax to do this, i.e.,:
```suggestion
        ..Default::default()
```

Reply:
Brilliant! Will do, for this and the item below.